src/shared/lib/StoreUtils.ts

```diff
@@ -1237,6 +1237,28 @@ export function groupBySampleId(
     }));
 }
 
+export function mapSampleIdToClinicalData<T extends { [key: string]: any }>(
```

Comment:
i might be confused but it seems strange that it's written as a generic and yet it it knows internally that it's dealing only with key to Array<ClinicalData>.

Reply:
i think this function doesn't need to accept indexKey or clinicalDatakey since it doesn't look like anything else is ever passed.    and i think it can be rewritten using lodash functions.   _.chain(arr).keyBy(item=>id).mapValues(item.clinicalDataKey)

Reply:
@sheridancbio double check .values()?

Reply:
My understanding (from reading about chain() here : https://lodash.com/docs/#chain is that after constructing the chain of functions to be applied, the output of the function chain must be "unwrapped" by calling .value(), so that addition to the suggested code looks good.

Since we know that all dictionaries passed in to this function are constructed with a key called 'id' and a key called 'clinicalData' associated with the types indicated, it makes sense to explicitly define the presence of these keys in the argument definition. (clinicalDataGroupedBySampleId)